libs/sumologic/native.go

```diff
@@ -128,11 +130,24 @@ func ConvertToSumoSLO(slo specs.OpenSLOSpec) (*SLO, error) {
 
 	indicator, _ := giveSLI(slo)
 
+	tagsMap := make(map[string]string)
+	for k := range slo.ObjectHeader.Metadata.Labels {
+		if len(slo.ObjectHeader.Metadata.Labels[k]) > 0 {
+			tagsMap[k] = slo.ObjectHeader.Metadata.Labels[k][0]
+		}
+	}
+
+	tagsStr, err := json.Marshal(tagsMap)
+	if err != nil {
+		return nil, err
+	}
+
 	sumoSLO := &SLO{
 		&sumotf.SLOLibrarySLO{
 			ResourceName:    resourceName,
 			Name:            slo.SLO.Metadata.Name,
 			Description:     slo.Spec.Description,
+			Tags:            strings.Replace(string(tagsStr), ":", " = ", -1),
```

Comment:
this does not seem right, because I don't see a restriction on OpenSLO tag values containing `:` character.

Reply:
I see. Do we allow `:` in sumo tags/values?

Reply:
We allow `=` in tag values. So if user has `:` in openSLO spec, we will create a valid sumo tag value. Irrespective of what we allow / not allow, the code here should not transform tag keys and tag values

Reply:
If we (i.,e native sumo) don't allow `:` in tag keys and values, we can add validation for it in slogen before the conversion.

If we allow `=`, then how would the terraform spec look like? For instance take the key-val pair `env : aws=2`. In this case, would the terraform spec be `env = aws=2`?

Reply:
> If we (i.,e native sumo) don't allow : in tag keys and values, we can add validation for it in slogen before the conversion.

Our validations will change in future. And we will likely introduce support for `:`. The code here should not depend on that. What you are suggesting will work (at run-time) in short term, but is still not solving the underlying code problem which is that it is potentially transforming keys and values supplied by user.

> If we allow =, then how would the terraform spec look like?

Likely enclosed within double quotes `env = "aws=2"`

Reply:
If we want to serialise tags as hcl string, we can just do it by hand instead of using json marshal ? That is, given a map[string]string, iterate over it and construct the string needed.

Reply:
yes, let me raise a new PR for it.

Reply:
fixed here https://github.com/OpenSLO/slogen/pull/67